cpp/modmesh/view/RMainWindow.cpp

```diff
@@ -30,11 +30,89 @@
 
 #include <modmesh/view/R3DWidget.hpp>
 
+#include <modmesh/view/RAction.hpp>
+#include <Qt>
+#include <QMenuBar>
+#include <QMenu>
+#include <QAction>
+#include <QActionGroup>
+
 namespace modmesh
 {
 
+RMainWindow::RMainWindow()
+    : QMainWindow()
+{
+    setUp();
+}
+
 void RMainWindow::setUp()
 {
+    this->setWindowIcon(QIcon(m_iconFilePath));
+
+    this->setMenuBar(new QMenuBar(nullptr));
+    m_fileMenu = this->menuBar()->addMenu(QString("File"));
+    m_appMenu = this->menuBar()->addMenu(QString("App"));
+    m_cameraMenu = this->menuBar()->addMenu(QString("Camera"));
+
+    {
+        auto * action = new RAction(
+            QString("New file"),
+            QString("Create new file"),
+            []()
+            {
+                qDebug() << "This is only a demo: Create new file!";
```

Comment:
A dummy file action.  Maybe it doesn't need to be kept.  Or it can really load a file.

Reply:
we can keep it, but we need a "save" button

Reply:
Sounds good.  But save what?

Reply:
we create a new file, and after editing, the user may want to save the file.

Reply:
But we have not created any new files yet.  What files do you think we will create in the future?

Reply:
Maybe I didn't say it clearly. I meant if there is a "create file" function, then there should also be a "save file" and "load file" function. Just like a normal editor?

Reply:
We do not have edit functionality at the moment. What would you anticipate the feature to edit?